torch/fx/experimental/symbolic_shapes.py

```diff
@@ -211,6 +225,28 @@ def bool_(self):
 
 
 if HAS_SYMPY:
+    # Overloaded to be compatible with regular Python.
+    # https://github.com/pytorch/pytorch/issues/90900
+    class Pow(sympy.Function):
+        @classmethod
+        def eval(cls, base, exp):
+            if exp == 0:
+                return sympy.Integer(1)
+            elif base == 0 and exp < 0:
+                raise ZeroDivisionError(f"{base} cannot be raised to a negative power")
+            else:
+                return base ** exp
```

Comment:
bleh, losing the printing here sucks

Reply:
cc @Chillee 

I think I'd be willing to "lie" and have these print as `**` and `/`

Reply:
When is printing an issue for these? These fully eval. Are you proposing not to eval the `else` cases and do the same as in `FloorDiv`?

Reply:
I don't think so. `truediv` is just this: `lambda a, b: TrueDiv(a, b)`, right? And `TrueDiv` is fully eval'd (on our side), so it dispatches to however `/` is handled. Seems fine to me:

```py
[ins] In [5]: TrueDiv(2,3)
Out[5]: 2/3

[ins] In [6]: Pow(2,3)
Out[6]: 8
```

Reply:
I'm talking about when we print expressions with symbolic variables in them

Reply:
Those work fine as well. Note that we dispatch to `**` or `/` in `else`, so they take care of printing.

```py
[ins] In [8]: TrueDiv(sympy.Symbol("y"),sympy.Symbol("x"))
Out[8]: y/x

[ins] In [9]: Pow(sympy.Symbol("y"),sympy.Symbol("x"))
Out[9]: y**x
```